### ISSUE TITLE
regexes doc doesn't mention $/ backreferences inside regex

### ISSUE THREAD

**Issue by timo** [perl6/doc#1545]
Not only is it important to know how exactly to refer back to already-matched things (though it's very simple in perl6), but it's very important to know that `$/` only updates whenever a `{ }` is encountered or when a `regex { }` or similar is finished

**Comment by lefth** [comment]
In particular, the docs don't say that you can't refer to an already-matched thing that has just been matched unless you force the named matches to update with `{ }`. But numbered matches do seem to work immediately.

**Comment by timo** [comment]
not only numbered matches, but also named matches. probably only references to $/ that are part of pieces of code, like `<subrule(Bool.pick ?? $0 !! "yo")>`

**Comment by JJ** [comment]
Let me see if I can come up with an example for this, as well as a reasonable use case.

**Comment by JJ** [comment]
I don't think I really understand what you propose. 
```perl6
say "zipi zape" ~~ /(\w)\w» \s+ \w+({$/})/;
```
This matches, but returns an empty match in the second case. Same if you use `$0`. Adding an empty block behind the first match:
```perl6
say "zipi zape" ~~ /(\w){}\w» \s+ \w+({$/})/;
```
Does not make any difference. This does not either
```perl6
say "zipi zape" ~~ /(\w){ $/ = $0 }\w» \s+ \w+({$/})/;
```

So what are you exactly proposing here?

**Comment by JJ** [comment]
We need to fix #2887 before this one.
